worlds/lingo/static_logic.py

```diff
@@ -197,363 +70,49 @@ def get_progressive_item_id(name: str):
-    else:
-        req_blocked_when_no_doors = False
-
-    required_door = None
-    if "required_door" in painting_data:
-        door = painting_data["required_door"]
-        required_door = RoomAndDoor(
-            door["room"] if "room" in door else room_name,
-            door["door"]
-        )
-
-    painting_obj = Painting(painting_id, room_name, enter_only, exit_only, orientation,
-                            required_painting, rwnd, required_door, disable_painting, move_painting, req_blocked,
-                            req_blocked_when_no_doors)
-    PAINTINGS[painting_id] = painting_obj
-    PAINTINGS_BY_ROOM[room_name].append(painting_obj)
-
-
-def process_progression(room_name, progression_name, progression_doors):
-    global PROGRESSIVE_ITEMS, PROGRESSION_BY_ROOM
-
-    # Progressive items are configured as a list of doors.
-    PROGRESSIVE_ITEMS.append(progression_name)
-
-    progression_index = 1
-    for door in progression_doors:
-        if isinstance(door, Dict):
-            door_room = door["room"]
-            door_door = door["door"]
-        else:
-            door_room = room_name
-            door_door = door
-
-        room_progressions = PROGRESSION_BY_ROOM.setdefault(door_room, {})
-        room_progressions[door_door] = Progression(progression_name, progression_index)
-        progression_index += 1
-
-
-def process_room(room_name, room_data):
-    global ROOMS, ALL_ROOMS
-
-    room_obj = Room(room_name, [])
-
-    if "entrances" in room_data:
-        for source_room, doors in room_data["entrances"].items():
-            process_entrance(source_room, doors, room_obj)
-
-    if "panels" in room_data:
-        PANELS_BY_ROOM[room_name] = dict()
-
-        for panel_name, panel_data in room_data["panels"].items():
-            process_panel(room_name, panel_name, panel_data)
-
-    if "doors" in room_data:
-        DOORS_BY_ROOM[room_name] = dict()
-
-        for door_name, door_data in room_data["doors"].items():
-            process_door(room_name, door_name, door_data)
-
-    if "paintings" in room_data:
-        PAINTINGS_BY_ROOM[room_name] = []
-
-        for painting_data in room_data["paintings"]:
-            process_painting(room_name, painting_data)
-
-    if "progression" in room_data:
-        for progression_name, progression_doors in room_data["progression"].items():
-            process_progression(room_name, progression_name, progression_doors)
+def load_static_data_from_file():
+    global HASHES, PAINTINGS, ALL_ROOMS, DOORS_BY_ROOM, PANELS_BY_ROOM, PROGRESSIVE_ITEMS, PROGRESSION_BY_ROOM, PAINTING_ENTRANCES, PAINTING_EXIT_ROOMS, PAINTING_EXITS, REQUIRED_PAINTING_ROOMS, REQUIRED_PAINTING_WHEN_NO_DOORS_ROOMS, SPECIAL_ITEM_IDS, PANEL_LOCATION_IDS, DOOR_LOCATION_IDS, DOOR_ITEM_IDS, DOOR_GROUP_ITEM_IDS, PROGRESSIVE_ITEM_IDS
+    
+    try:
+        from importlib.resources import files
+    except ImportError:
+        from importlib_resources import files
```

Comment:
Hey can we drop 3.8 already

Also, here's the way I would do this, no clue if this is actually "better"
```
    file = pkgutil.get_data(__name__, "data/generated.dat")

    if file is None:
        return

    pickdata = pickle.load(BytesIO(file))
```

Reply:
I mean idk it's probably slower because you're loading the data and then feeding it into a BytesIO to load it again, the fcking ImportError thing is just so god damn ugly lol